### PR TITLE
Complete removal of MapCube

### PR DESCRIPTION
**Pull request by MSKirk**
Removes MapCube and references to MapCube. This is a complement to [PR 2603](https://github.com/sunpy/sunpy/pull/2603) to complete the transition from MapCube to MapSequence.

### CODE REVIEW

File: docs/guide/data_types/maps.rst [hunk 1]
```diff
@@ -558,39 +558,39 @@ or by providing a directory full of image files::
 
     >>> mc = sunpy.map.Map('path/to/my/files/*.fits', cube=True)   #  doctest: +SKIP
```
Comment (vn-ki) [inline docs/guide/data_types/maps.rst]: This and above line should read `sequence=True` instead of `cube=True`

File: sunpy/instr/iris/iris.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 def SJI_to_cube(filename, start=0, stop=None, hdu=0):
```
Comment (vn-ki) [inline sunpy/instr/iris/iris.py]: Name remains `SJI_to_cube`?

**Comment by MSKirk** [comment]
That makes sense. It sounded like a simple fix when I first started it...

**Comment by nabobalis** [comment]
It's our fault for leaving for so long and not getting to this PR sooner.

**Pull request closed without merging**